In [1]:
'''
Sources:
sentiment analysis inspiration: https://www.kaggle.com/code/ekrembayar/f-r-i-e-n-d-s-text-mining-data-visualization
AFINN docs: https://github.com/fnielsen/afinn
sqlite doc: https://www.sqlite.org/
Genius API docs: https://docs.genius.com/
Genius API tutorial: https://www.youtube.com/watch?v=cU8YH2rhN6A

'''

import os
import lyricsgenius
import json
from dotenv import load_dotenv
from pprint import pprint
import sqlite3
from afinn import Afinn
import re

afinn = Afinn()

load_dotenv()

genius_access_token = os.getenv('GENIUS_ACCESS_TOKEN')

genius = lyricsgenius.Genius(genius_access_token)

conn = sqlite3.connect('songs.db')
cursor = conn.cursor()

cursor.execute(
    '''
    CREATE TABLE IF NOT EXISTS songs (
        id INTEGER PRIMARY KEY,
        title TEXT NOT NULL,
        artist TEXT NOT NULL,
        genre TEXT,
        date DATE,
        country TEXT,
        lyrics TEXT,
        sentiment REAL
    )
    '''
)

def addSong(title, artist, genre, date, country):
    print(f"Adding: {title}, {artist}, {genre}, {date}, {country}")

    lyrics = genius.search_song(title, artist)

    if lyrics:
        lyrics = str(lyrics.lyrics)
        lyrics.replace("'", "''")
        lyrics = re.sub(r'[^a-zA-Z0-9\s]', '', lyrics)
        print(f"Lyrics: {lyrics}")
        sentiment = afinn.score(lyrics)
        print(f"Sentiment Score: {sentiment}")
        cursor.execute(f'''
            INSERT INTO songs (title, artist, genre, date, country, lyrics, sentiment)
            VALUES ('{title}', '{artist}', '{genre}', {date}, '{country}', '{lyrics}', {sentiment})
        ''')
    else:
        print("Song not found by Genius API.")





In [4]:

addSong("Butterfly Effect", "Travis Scott", "Rap", "2017", "USA")


Adding: Butterfly Effect, Travis Scott, Rap, 2017, USA
Searching for "Butterfly Effect" by Travis Scott...
Done.
Lyrics: 273 ContributorsTranslationsPortugusEspaolTrkeItalianoSvenskaFranaisBUTTERFLY EFFECT LyricsIntro
All the commas
Murda on the beat so its not nice
Ooh hmm

Chorus
For this life I cannot change Change
Hidden Hills deep off in the main Main
MMs sweet like candy cane Cane
Drop the top pop it let it bang Pop it pop it
For this life I cannot change
In the hills deep off in the main
MMs sweet like candy cane
Drop the top pop it let it bang Pop it pop it

Verse 1
Drop the top play hide and seek Yeah
Jump inside jump straight to the league League
Take a sip feel just how I be Its lit
On Freeway but no aint nothin free Straight up
Bend laws bend lanes Skrrt skrrt
Been bustin bills but still aint nothin change Skrrt skrrt skrrt skrrt
You in the mob soon as you rock the chain Skrrt skrrt mob
She caught the waves just thumbin through my braids Alright
Heatin up baby Im just heati

# Output

Adding: Butterfly Effect, Travis Scott, Rap, 2017, USA
Searching for "Butterfly Effect" by Travis Scott...
Done.
Lyrics: 273 ContributorsTranslationsPortugusEspaolTrkeItalianoSvenskaFranaisBUTTERFLY EFFECT LyricsIntro
All the commas
Murda on the beat so its not nice
Ooh hmm

Chorus
For this life I cannot change Change
Hidden Hills deep off in the main Main
MMs sweet like candy cane Cane
Drop the top pop it let it bang Pop it pop it
For this life I cannot change
In the hills deep off in the main
MMs sweet like candy cane
Drop the top pop it let it bang Pop it pop it

Verse 1
Drop the top play hide and seek Yeah
Jump inside jump straight to the league League
Take a sip feel just how I be Its lit
On Freeway but no aint nothin free Straight up
Bend laws bend lanes Skrrt skrrt
Been bustin bills but still aint nothin change Skrrt skrrt skrrt skrrt
...
Hidden Hills deep off in the main
MMs sweet like candy cane
Drop the top pop it let it bang Bang yeah215Embed
Sentiment Score: 62.0